core/src/main/java/io/undertow/io/UndertowOutputStream.java

```diff
@@ -366,4 +380,12 @@ private ByteBuffer buffer() {
         return this.buffer;
     }
 
+    private long writeBlocking(StreamSinkChannel channel, ByteBuffer[] buffers, int offs, int len) throws IOException {
+        if (writeTimeoutMillis > 0) {
+            return BlockingChannels.writeBlockingOrThrow(
```

Comment:
bug: this should internally respect the write timeout, but loop as long as bytes are successfully written. Existing callers ignore the result of this method assuming all bytes are fully written.